torchvision/datasets/video_utils.py

```diff
@@ -255,6 +255,50 @@ def get_clip_location(self, idx):
             clip_idx = idx - self.cumulative_sizes[video_idx - 1]
         return video_idx, clip_idx
 
+    def get_idx_range_of_video(self, idx):
+        """
+        Provided an idx of a clip, the function returns the indices range belong to the same video
+        """
+
+        video_idx = bisect.bisect_right(self.cumulative_sizes, idx)
+        if video_idx == 0:
+            idx_start = 0
+        else:
+            idx_start = self.cumulative_sizes[video_idx - 1]
+
+        if len(self.cumulative_sizes) == 0:
+            idx_end = 0
+        else:
+            idx_end = self.cumulative_sizes[video_idx] - 1
+
+        return idx_start, idx_end
+
+    def get_video_path(self, idx):
+        """
+        Converts a flattened representation of the indices into a video path
+        """
+        video_idx = bisect.bisect_right(self.cumulative_sizes, idx)
+        return self.video_paths[video_idx]
+
+    def get_video_name(self, idx):
```

Comment:
Does this add extra information than just passing the full `video_path`? I think just returning the full filename would be enough, as done by `get_video_path`, so I'd say that we could remove this method.

But then, `video_idx` is already returned by `get_clip`, so the user can retrieve this information as well without having to call an extra method

Reply:
I agree. This information can be easily retrieved the way I did. However, ithe method might be helpful if one would be interesed in the video name containing a specific clip without reading the code of videoClips.

Reply:
ok. Let's just return the full video path, instead of the name, what do you think?

Reply:
Sounds good.